### PR TITLE
Add ctx.Scheme() and update ctx.Protocol() methods [🎌 breaking change]

### LINKED ISSUE #1712: 🚀 c.Scheme() and c.Protocol()
### Feature 1
```go
	app.Get("/", func(c *Ctx) error {
		return c.JSON(c.Scheme()) // return "http" or "https"
	})
```
`c.Scheme()` is going to return fasthttp request uri's scheme.

---
### Feature 2 (Actually Update)
```go
	app.Get("/", func(c *Ctx) error {
		return c.JSON(c.Protocol()) // return "HTTP/1.1" or "HTTP/2.0"
	})
```
`c.Protocol()` return `http` or `https` normally. I think, c.Protocol should return `HTTP/1.1` or `HTTP/2.0`. Because fasthttp request header's protocol return `HTTP/1.1.` or `HTTP/2.0`. 

If you think like me, I can help for PR.

### PR DESCRIPTION
**Pull request by abhi12299**
**Please provide enough information so that others can review your pull request:**

Updating the `ctx.Protocol` method to return HTTP protocol (like HTTP/1.0, HTTP/1.1) and changing `ctx.Scheme` method to return the request scheme (like http, https).

**Explain the *details* for making this change. What existing problem does the pull request solve?**
Closes #1712 

`c.Scheme()` usage:
```go
app.Get("/", func(c *Ctx) error {
    return c.JSON(c.Scheme()) // return "http" or "https"
})
```

`c.Protocol()` usage:
```go
app.Get("/", func(c *Ctx) error {
    return c.JSON(c.Protocol()) // return "HTTP/1.1"
})
```

### CODE REVIEW

File: ctx.go [hunk 1]
```diff
@@ -848,9 +848,9 @@ func (c *Ctx) Path(override ...string) string {
 	return c.path
 }
 
-// Protocol contains the request protocol string: http or https for TLS requests.
+// Scheme contains the request scheme: http or https for TLS requests.
 // Use Config.EnableTrustedProxyCheck to prevent header spoofing, in case when your app is behind the proxy.
-func (c *Ctx) Protocol() string {
+func (c *Ctx) Scheme() string {
```
Comment (efectn) [inline ctx.go:853]: This is breaking change i think.

Comment (abhi12299) [reply ctx.go:853]: Yes, it is. I raised this in the discussion as well. The tests have all passed though.

**Comment by ReneWerner87** [comment]
@abhi12299 could you create a PR for our doc repository

- [ ] update documentation repo

**Comment by ReneWerner87** [comment]
@abhi12299 
image

pls check the logging middleware

**Comment by abhi12299** [comment]
@ReneWerner87 will do.. thanks!

**Comment by ReneWerner87** [comment]
https://github.com/gofiber/contrib/blob/d5ab16384e7b063668b9a23ca9aa234dc856132c/fiberzap/zap.go#L81
https://github.com/gofiber/contrib/blob/9a79234709e3a33fa3213e76536be90559c18fa0/otelfiber/fiber.go#L75
https://github.com/gofiber/recipes/blob/dbc34fbfa3706144d54089d79c6505e9e95c81b3/https-tls/main.go#L20
https://github.com/gofiber/fiber/blob/6036f402dbddee9bf6e40e41c8a738da54d1c55a/middleware/logger/logger.go#L214

File: ctx.go [hunk 2]
```diff
@@ -1415,3 +1415,8 @@ func (c *Ctx) IsFromLocal() bool {
 	}
 	return c.isLocalHost(ips[0])
 }
+
+// Protocol returns the request's HTTP protocol: HTTP/1.1,  HTTP/1.0
+func (c *Ctx) Protocol() string {
+	return string(c.fasthttp.Request.Header.Protocol())
```
Comment (balcieren) [inline ctx.go:1421]: @abhi12299 pls use this https://github.com/gofiber/fiber/blob/master/utils/convert.go#L16

Comment (abhi12299) [reply ctx.go]: Done. Check: [9667849](https://github.com/gofiber/fiber/pull/1746/commits/96678497eba1722057e63d5b0082df9fbaf299c6)

File: ctx.go (hunk 1 above)
Comment (balcieren) [inline ctx.go:853]: @abhi12299 can you look c.Request().URI().Scheme(), it could return `http` or `https`

Comment (abhi12299) [reply ctx.go:853]: @balcieren I haven't changed the behavior of the old `Protocol` function.. just renamed it to `Scheme`. So it works just like it used to previously.

**Comment by abhi12299** [comment]
> @abhi12299 could you create a PR for our doc repository
> 
> * [ ]  update documentation repo

Hey, I've raised a PR for this: [#237](https://github.com/gofiber/docs/pull/237)

**Comment by abhi12299** [comment]
> zap: https://github.com/gofiber/contrib/blob/d5ab16384e7b063668b9a23ca9aa234dc856132c/fiberzap/zap.go#L81
> 
> otelfiber: https://github.com/gofiber/contrib/blob/9a79234709e3a33fa3213e76536be90559c18fa0/otelfiber/fiber.go#L75
> 
> reciepes: https://github.com/gofiber/recipes/blob/dbc34fbfa3706144d54089d79c6505e9e95c81b3/https-tls/main.go#L20
> 
> logger:
> 
> https://github.com/gofiber/fiber/blob/6036f402dbddee9bf6e40e41c8a738da54d1c55a/middleware/logger/logger.go#L214

@ReneWerner87 how do I run tests with the updated `Scheme` and `Protocol` methods in zap, otelfiber and reciepes? It would fail as of now.

**Comment by ReneWerner87** [comment]
> > zap: https://github.com/gofiber/contrib/blob/d5ab16384e7b063668b9a23ca9aa234dc856132c/fiberzap/zap.go#L81
> > otelfiber: https://github.com/gofiber/contrib/blob/9a79234709e3a33fa3213e76536be90559c18fa0/otelfiber/fiber.go#L75
> > reciepes: https://github.com/gofiber/recipes/blob/dbc34fbfa3706144d54089d79c6505e9e95c81b3/https-tls/main.go#L20
> > logger:
> > https://github.com/gofiber/fiber/blob/6036f402dbddee9bf6e40e41c8a738da54d1c55a/middleware/logger/logger.go#L214
> 
> @ReneWerner87 how do I run tests with the updated `Scheme` and `Protocol` methods in zap, otelfiber and recipes? It would fail as of now.

we will do it after the release in the next week

**Comment by ReneWerner87** [comment]
after closer examination and consultation with the other project members, we will probably adopt this change only in v3.

**Comment by ReneWerner87** [comment]
Thanks for your work

too bad that this decision was made by us only later

**Pull request closed without merging**